tests/network/vmi_multus.go

```diff
@@ -335,26 +356,59 @@ var _ = SIGDescribe("[Serial]Multus", func() {
 		})
 
 		Context("VirtualMachineInstance with Linux bridge plugin interface", func() {
-			table.DescribeTable("should be able to ping between two vms", func(interfaces []v1.Interface, networks []v1.Network, ifaceName string) {
+			getIfaceIPByNetworkName := func(vmiName, networkName string) (string, error) {
+				vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Get(vmiName, &metav1.GetOptions{})
+				if err != nil {
+					return "", err
+				}
+
+				for _, iface := range vmi.Status.Interfaces {
+					if iface.Name == networkName {
+						return iface.IP, nil
+					}
+				}
+
+				return "", fmt.Errorf("couldn't find iface %s on vmi %s", networkName, vmiName)
+			}
+
+			generateIPAMConfig := func(ipamType string, subnet string) string {
+				return fmt.Sprintf("\\\"type\\\": \\\"%s\\\", \\\"subnet\\\": \\\"%s\\\"", ipamType, subnet)
+			}
+
+			table.DescribeTable("should be able to ping between two vms", func(interfaces []v1.Interface, networks []v1.Network, ifaceName, staticIPVm1, staticIPVm2 string) {
+				if staticIPVm2 == "" || staticIPVm1 == "" {
+					ipam := generateIPAMConfig("host-local", "10.1.1.0/24")
+					Expect(createBridgeNetworkAttachementDefinition("linux-bridge-net-ipam", 0, ipam)).To(Succeed())
+				}
+
 				vmiOne := createVMIOnNode(interfaces, networks)
 				vmiTwo := createVMIOnNode(interfaces, networks)
 
 				tests.WaitUntilVMIReady(vmiOne, console.LoginToAlpine)
 				tests.WaitUntilVMIReady(vmiTwo, console.LoginToAlpine)
 
-				Expect(configInterface(vmiOne, ifaceName, "10.1.1.1/24")).To(Succeed())
+				Expect(configureAlpineInterfaceIP(vmiOne, ifaceName, staticIPVm1)).To(Succeed())
 				By(fmt.Sprintf("checking virtual machine interface %s state", ifaceName))
 				Expect(checkInterface(vmiOne, ifaceName)).To(Succeed())
 
-				Expect(configInterface(vmiTwo, ifaceName, "10.1.1.2/24")).To(Succeed())
+				Expect(configureAlpineInterfaceIP(vmiTwo, ifaceName, staticIPVm2)).To(Succeed())
 				By(fmt.Sprintf("checking virtual machine interface %s state", ifaceName))
 				Expect(checkInterface(vmiTwo, ifaceName)).To(Succeed())
 
+				ipAddr, err := cidrToIP(staticIPVm2)
```

Comment:
Shouldn't this be done only if `staticIPVm2 !=""`?

Reply:
Right! Thanks for the catch.
Done